File: SDKs/collegefootballdata.com/4.1.6/python/src/sdk/players.py
import requests
from typing import List,Optional
from sdk.models import shared, operations
from . import utils

class Players:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def get_player_season_stats(self, request: operations.GetPlayerSeasonStatsRequest) -> operations.GetPlayerSeasonStatsResponse:
        r"""Player stats by season
        Season player stats
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/stats/player/season"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.GetPlayerSeasonStatsResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[List[shared.PlayerSeasonStat]])
                res.player_season_stats = out
        elif r.status_code == 400:
            pass

        return res

    
    def get_player_usage(self, request: operations.GetPlayerUsageRequest) -> operations.GetPlayerUsageResponse:
        r"""Player usage metrics broken down by season
        Player usage metrics by season
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/player/usage"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.GetPlayerUsageResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[List[shared.PlayerUsage]])
                res.player_usages = out
        elif r.status_code == 400:
            pass

        return res

    
    def get_returning_production(self, request: operations.GetReturningProductionRequest) -> operations.GetReturningProductionResponse:
        r"""Team returning production metrics
        Returning production metrics
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/player/returning"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.GetReturningProductionResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[List[shared.ReturningProduction]])
                res.returning_productions = out
        elif r.status_code == 400:
            pass

        return res

    
    def player_search(self, request: operations.PlayerSearchRequest) -> operations.PlayerSearchResponse:
        r"""Search for player information
        Search for players
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/player/search"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.PlayerSearchResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[List[shared.PlayerSearchResult]])
                res.player_search_results = out
        elif r.status_code == 400:
            pass

        return res

    

File: SDKs/collegefootballdata.com/4.1.6/go/players.go
package sdk

import (
	"context"
	"fmt"
	"net/http"
	"openapi/internal/utils"
	"openapi/pkg/models/operations"
	"openapi/pkg/models/shared"
	"strings"
)

type Players struct {
	_defaultClient  HTTPClient
	_securityClient HTTPClient
	_serverURL      string
	_language       string
	_sdkVersion     string
	_genVersion     string
}

func NewPlayers(defaultClient, securityClient HTTPClient, serverURL, language, sdkVersion, genVersion string) *Players {
	return &Players{
		_defaultClient:  defaultClient,
		_securityClient: securityClient,
		_serverURL:      serverURL,
		_language:       language,
		_sdkVersion:     sdkVersion,
		_genVersion:     genVersion,
	}
}

// GetPlayerSeasonStats - Player stats by season
// Season player stats
func (s *Players) GetPlayerSeasonStats(ctx context.Context, request operations.GetPlayerSeasonStatsRequest) (*operations.GetPlayerSeasonStatsResponse, error) {
	baseURL := s._serverURL
	url := strings.TrimSuffix(baseURL, "/") + "/stats/player/season"

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("error creating request: %w", err)
	}

	utils.PopulateQueryParams(ctx, req, request.QueryParams)

	client := s._securityClient

	httpRes, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("error sending request: %w", err)
	}
	defer httpRes.Body.Close()

	contentType := httpRes.Header.Get("Content-Type")

	res := &operations.GetPlayerSeasonStatsResponse{
		StatusCode:  int64(httpRes.StatusCode),
		ContentType: contentType,
	}
	switch {
	case httpRes.StatusCode == 200:
		switch {
		case utils.MatchContentType(contentType, `application/json`):
			var out []shared.PlayerSeasonStat
			if err := utils.UnmarshalJsonFromResponseBody(httpRes.Body, &out); err != nil {
				return nil, err
			}

			res.PlayerSeasonStats = out
		}
	case httpRes.StatusCode == 400:
	}

	return res, nil
}

// GetPlayerUsage - Player usage metrics broken down by season
// Player usage metrics by season
func (s *Players) GetPlayerUsage(ctx context.Context, request operations.GetPlayerUsageRequest) (*operations.GetPlayerUsageResponse, error) {
	baseURL := s._serverURL
	url := strings.TrimSuffix(baseURL, "/") + "/player/usage"

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("error creating request: %w", err)
	}

	utils.PopulateQueryParams(ctx, req, request.QueryParams)

	client := s._securityClient

	httpRes, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("error sending request: %w", err)
	}
	defer httpRes.Body.Close()

	contentType := httpRes.Header.Get("Content-Type")

	res := &operations.GetPlayerUsageResponse{
		StatusCode:  int64(httpRes.StatusCode),
		ContentType: contentType,
	}
	switch {
	case httpRes.StatusCode == 200:
		switch {
		case utils.MatchContentType(contentType, `application/json`):
			var out []shared.PlayerUsage
			if err := utils.UnmarshalJsonFromResponseBody(httpRes.Body, &out); err != nil {
				return nil, err
			}

			res.PlayerUsages = out
		}
	case httpRes.StatusCode == 400:
	}

	return res, nil
}

// GetReturningProduction - Team returning production metrics
// Returning production metrics
func (s *Players) GetReturningProduction(ctx context.Context, request operations.GetReturningProductionRequest) (*operations.GetReturningProductionResponse, error) {
	baseURL := s._serverURL
	url := strings.TrimSuffix(baseURL, "/") + "/player/returning"

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("error creating request: %w", err)
	}

	utils.PopulateQueryParams(ctx, req, request.QueryParams)

	client := s._securityClient

	httpRes, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("error sending request: %w", err)
	}
	defer httpRes.Body.Close()

	contentType := httpRes.Header.Get("Content-Type")

	res := &operations.GetReturningProductionResponse{
		StatusCode:  int64(httpRes.StatusCode),
		ContentType: contentType,
	}
	switch {
	case httpRes.StatusCode == 200:
		switch {
		case utils.MatchContentType(contentType, `application/json`):
			var out []shared.ReturningProduction
			if err := utils.UnmarshalJsonFromResponseBody(httpRes.Body, &out); err != nil {
				return nil, err
			}

			res.ReturningProductions = out
		}
	case httpRes.StatusCode == 400:
	}

	return res, nil
}

// PlayerSearch - Search for player information
// Search for players
func (s *Players) PlayerSearch(ctx context.Context, request operations.PlayerSearchRequest) (*operations.PlayerSearchResponse, error) {
	baseURL := s._serverURL
	url := strings.TrimSuffix(baseURL, "/") + "/player/search"

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("error creating request: %w", err)
	}

	utils.PopulateQueryParams(ctx, req, request.QueryParams)

	client := s._securityClient

	httpRes, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("error sending request: %w", err)
	}
	defer httpRes.Body.Close()

	contentType := httpRes.Header.Get("Content-Type")

	res := &operations.PlayerSearchResponse{
		StatusCode:  int64(httpRes.StatusCode),
		ContentType: contentType,
	}
	switch {
	case httpRes.StatusCode == 200:
		switch {
		case utils.MatchContentType(contentType, `application/json`):
			var out []shared.PlayerSearchResult
			if err := utils.UnmarshalJsonFromResponseBody(httpRes.Body, &out); err != nil {
				return nil, err
			}

			res.PlayerSearchResults = out
		}
	case httpRes.StatusCode == 400:
	}

	return res, nil
}
